### DISCUSSION TITLE
Factorial

### DISCUSSION THREAD

**Discussion by naomipoclava2021** [santiemanuel/milProgramadoresQA#6] [Creaciones]
Asi es posible anidar un ciclo for dentro de un while.  Factorial=1*2*3*4*5*6*....*
```python
##Este programa calcula en factorial de un numero

mensaje="si"
while(mensaje=="si"):
  mensaje=input("Deasea continuar: ")
  if(mensaje=="si"):
    numero=int(input("Ingresa  un numero: "))
    factorial=1
    for i in range(1,numero+1):
      factorial=factorial * i
    print("El factorial es: ",factorial)
  elif(mensaje=="no"):
    mansaje="no"
    print("Fin del programa")
  else:
    print("ingrese una opcion valida")
```

**Comment by santiemanuel** [comment]
Hola, en la parte del elif del código se lo puede optimizar.

- Ya se pregunta si mensaje es "no" así que asignarle "no" en el verdadero no le va cambiar el valor.
- Si es cualquier otra cosa que no sea "si" o "no" el programa mostrará que se debe ingresar una opción válida.

Pero acá hay un problema, se ejecuta el código del else y sube a evaluar la condición del while, al no cumplirse el ciclo terminará.

Para darle sentido al mensaje de "Ingresar una opción válida" tenés que hacer otro ciclo interior de validación de la entrada hasta que se ingrese "si" o "no".
